com.reprezen.swagedit/src/com/reprezen/swagedit/editor/hyperlinks/JsonReferenceHyperlinkDetector.java

```diff
@@ -42,109 +35,56 @@
-		if (info.text.matches(LOCAL_REF_PATTERN.pattern())) {
-			return doDetectLocalLink(doc, viewer, info);
-		} else {
-			return doDetectExternalLink(info);
-		}
-	}
-
-	private IHyperlink[] doDetectExternalLink(HyperlinkInfo info) {
-		String filePath;
-		String pointer = null;
-		IRegion target = null;
-
-		if (info.text.contains("#")) {
-			filePath = sanitize(info.text.split("#")[0]);
-			pointer = sanitize(info.text.split("#")[1]);
-		} else {
-			filePath = sanitize(info.text);
-		}
+		URI baseURI = getBaseURI();
 
-		IFile file = getFile(filePath);
-		if (file == null || !file.exists()) {
+		JsonReference reference = getFactory().create(doc.getNodeForPath(basePath));
+		if (!reference.isValid(baseURI)) {
 			return null;
 		}
 
-		if (pointer != null) {
-			SwaggerDocument doc = getExternalDocument(file);
-			if (doc != null) {
-				target = doc.getRegion(asPath(pointer));
+		if (reference.isLocal()) {
+			IRegion target = doc.getRegion(pointer(reference.getPointer()));
+			if (target == null) {
+				return null;
 			}
-		}
-
-		return new IHyperlink[] { new SwaggerFileHyperlink(info.region, info.text, file, target) };
-	}
-
-	private IHyperlink[] doDetectLocalLink(SwaggerDocument doc, ITextViewer viewer, HyperlinkInfo info) {
-		Matcher matcher = LOCAL_REF_PATTERN.matcher(info.text);
-		String pointer = null;
-		if (matcher.find()) {
-			pointer = matcher.group(1);
-		}
-
-		IRegion target = doc.getRegion(asPath(pointer));
-		if (target == null) {
-			return null;
-		}
-
-		return new IHyperlink[] { new SwaggerHyperlink(pointer, viewer, info.region, target) };
-	}
-
-	private IFile getFile(String filePath) {
-		IEditorInput input = PlatformUI.getWorkbench()
-				.getActiveWorkbenchWindow()
-				.getActivePage()
-				.getActiveEditor()
-				.getEditorInput();
-
-		if (input instanceof FileEditorInput) {
-			FileEditorInput fileInput = (FileEditorInput) input;
-
-			IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
-			IPath extPath = new Path(filePath);
-			if (!extPath.isAbsolute()) {
-				extPath = new Path(fileInput.getURI().resolve(extPath.toOSString()).getPath());
+			return new IHyperlink[] {
+					new SwaggerHyperlink(reference.getPointer().toString(), viewer, info.region, target) };
+		} else {
+			IFile file = DocumentUtils.getWorkspaceFile(baseURI.resolve(reference.getUri()));
+			if (file != null && file.exists()) {
+				return new IHyperlink[] {
+						new SwaggerFileHyperlink(info.region, info.text, file, pointer(reference.getPointer())) };
 			}
-
-			return root.getFileForLocation(extPath);
 		}
 
 		return null;
 	}
 
-	private SwaggerDocument getExternalDocument(IFile file) {
-		final SwaggerDocument doc = new SwaggerDocument();
-		try {
-			doc.set(CharStreams.toString(new InputStreamReader(file.getContents())));
-		} catch (IOException | CoreException e) {
-			return null;
-		}
-
-		return doc;
+	protected FileEditorInput getActiveEditor() {
+		return DocumentUtils.getActiveEditorInput();
```

Comment:
Do we need this method if it simply returns `DocumentUtils.getActiveEditorInput()`?